app/Models/Collections/Artwork.php

```diff
@@ -610,4 +620,21 @@ public function docExtra()
 
         return $doc;
     }
+
+    public function toSearchableArray()
+    {
+        $array = $this->transform();
+        $keysToUnset = [];
+        foreach ($array as $key => $value) {
+            if (empty($value)) {
+                $keysToUnset[] = $key;
+            }
+        }
+
+        foreach ($keysToUnset as $key) {
+            unset($array[$key]);
+        }
+
+        return $array;
```

Comment:
This could be simplified:
```php
return array_filter($this->transform());
```

See https://www.php.net/manual/en/function.array-filter.php